rust/index/src/hnsw_provider.rs

```diff
@@ -376,7 +343,17 @@ impl HnswIndexProvider {
                 }
                 Err(e) => Err(Box::new(HnswIndexProviderOpenError::IndexLoadError(e))),
             },
-        }
+        };
+
+        // Cleanup directory.
+        Self::purge_one_id(&self.temporary_storage_path, *id)
```

Comment:
Do we care that this is happening in the foreground?

Reply:
should be much faster compared to the overall load time